ludwig/backend/ray.py

```diff
@@ -176,16 +177,20 @@ def batch_predict(self, model, dataset, *args, **kwargs):
         predictor_kwargs = self.predictor_kwargs
         output_columns = get_output_columns(model.output_features)
 
-        def batch_predict_partition(dataset):
+        def batch_predict_partition(df):
```

Comment:
This will be more efficient if we turn it into a (stateful) actor:

```
class BatchInferModel:
    def __init__(self):
        self.model = remote_model.load()
        self.predictor = Predictor(**predictor_kwargs)
    
    def __call__(self, df: pd.DataFrame) -> pd.DataFrame:
        pd_ds = PandasDataset(df, dataset.features, dataset.data_hdf5_fp)
        predictions = self.predictor.batch_predict(self.model, pd_ds, *args, **kwargs)
        ordered_predictions = predictions[output_columns]
        return ordered_predictions

return dataset.ds.map_batchest(BatchInferModel, ...)
```

This allows us to cache the model and predictor in memory between calls.

Reply:
Thanks for the suggestion, done!